utils/types.ts

```diff
@@ -0,0 +1,11 @@
+export type bookingType = {
+  id: number;
+  title: string;
+  firstName: string;
+  surname: string;
+  email: string;
+  roomId: string;
+  checkInDate: string;
+  checkOutDate: string;
+  [key: string]: string | number;
```

Comment:
we can remove this line to keep `bookingType` more strictly typed

Reply:
You sure???????????
<img width="642" alt="Screenshot 2023-04-22 at 01 20 03" src="https://user-images.githubusercontent.com/107769663/233751662-fafb5dd2-cbae-4b33-ac8a-e906398ee60d.png">

Reply:
if I don't use `[key: string]: string | number;`, it will report an error⋯

Reply:
This is what I mean:

```
type KeyType = "firstName" | "surname" | "email";
  
if (req.query.term) {
  const keys: KeyType[] = ["firstName", "surname", "email"];
  const filtered = bookings.filter((booking) =>
    keys.some((key) =>
      (booking[key] as string)
        .toLowerCase()
        .includes(req.query.term!.toString().toLowerCase())
    )
  );
  return res.send(filtered);
}
```

Reply:
It turns out that keys also need types..., and it can even remove 'as String'....

Reply:
yes you can remove `as string`. This can make keys more strictly typed, prevent errors caused by unexpected key string